sql/core/src/test/scala/org/apache/spark/sql/parquet/ParquetIOSuite.scala

```diff
@@ -329,6 +329,16 @@ class ParquetIOSuiteBase extends QueryTest with ParquetTest {
       checkAnswer(parquetFile(file), (data ++ newData).map(Row.fromTuple))
     }
   }
+
+  test("read non-existent parquet file") {
+    intercept[java.io.FileNotFoundException] {
```

Comment:
Would be good to add a comment about the original behavior, since the intention of this test case is not super intuitive.